backend/file/file_test.go

```diff
@@ -90,17 +90,27 @@ timeout = 10
 		defer cleanup()
 
 		var c config
-		b := file.NewBackend(path)
+		b := file.NewBackend(path, false)
 
 		err := b.Unmarshal(context.Background(), &c)
 		require.Error(t, err)
 	})
 
-	t.Run("File not found", func(t *testing.T) {
+	t.Run("Required file not found", func(t *testing.T) {
 		var c config
-		b := file.NewBackend("some path")
+		b := file.NewBackend("some path", false)
 
 		err := b.Unmarshal(context.Background(), &c)
 		require.Error(t, err)
 	})
+
+	t.Run("Optional file not found", func(t *testing.T) {
+		var c config
+		b := file.NewBackend("some path", true)
+
+		err := b.Unmarshal(context.Background(), &c)
+		require.Error(t, err)
+		_, ok := err.(file.ErrOpenOptionalFile)
```

Comment:
This doesn't work: https://play.golang.org/p/rl1ShVHrEFw
It works in your test, but in fact, any error will return true

Reply:
Oh yeah because in fact `ErrOpenOptionalFile` is just an alias to the `error` built-in interface, right?

Reply:
Done in ad90a5e